modules/nw-route-specific-annotations.adoc

```diff
@@ -188,3 +188,16 @@ The following table provides examples of the path rewriting behavior for various
 |/foo/|/foo/|/|/
 |/foo/|/foo/bar|/|/bar
 |===
+
+Certain special characters in `haproxy.router.openshift.io/rewrite-target` require special handling because they must be escaped properly. Refer to the following table to understand how these characters are handled.
+
+.Special character handling:
+[cols="3*", options="header"]
+|===
+|For character|Use characters|Notes
+|#|\#|Avoid # because it terminates the regular expression
```

Comment:
I realized we mention "regular expression". I don't think users will know what that means since that's an implementation detail. Maybe it makes sense to make it more generic.

```suggestion
|#|\#|Avoid # because it terminates the rewrite expression
```